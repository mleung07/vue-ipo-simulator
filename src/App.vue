<template>
  <div id="app">
    <TopBar />
    <Calculator />
    <Disclaimer />
  </div>
</template>

<script>
import TopBar from "./components/TopBar";
import Calculator from "./components/Calculator";
import Disclaimer from "./components/Disclaimer";

export default {
  name: "App",
  components: {
    TopBar,
    Calculator,
    Disclaimer,
  },
};
</script>

<style>
#app {
  font-family: "Roboto", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
body {
  margin: 0;
}
</style>
