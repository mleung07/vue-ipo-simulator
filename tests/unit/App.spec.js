import { mount } from "@vue/test-utils";
import App from "@/App.vue";

describe("App.vue", () => {
  it("matches snapshot", () => {
    const wrapper = mount(App);
    expect(wrapper.element).toMatchSnapshot();
  });
});
