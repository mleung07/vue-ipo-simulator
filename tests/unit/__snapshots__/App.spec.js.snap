// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches snapshot 1`] = `
<div
  id="app"
>
  <div
    class="header"
  >
    <div
      class="title"
    >
      IPO Simulator
    </div>
  </div>
   
  <div
    id="calculator"
  >
    <div
      class="input-box"
    >
      <input
        required="required"
        type="text"
      />
       
      <label>
        Shares Appiled
      </label>
    </div>
     
    <div
      class="input-box"
    >
      <input
        required="required"
        type="text"
      />
       
      <label>
        Offering Price
      </label>
    </div>
     
    <div
      class="input-box"
    >
      <input
        required="required"
        type="text"
      />
       
      <label>
        Shares Allocated
      </label>
    </div>
     
    <div
      class="input-box"
    >
      <input
        required="required"
        type="text"
      />
       
      <label>
        Selling Price
      </label>
    </div>
     
    <div
      class="input-box"
    >
      <input
        required="required"
        type="text"
      />
       
      <label>
        Handling Fee
      </label>
    </div>
     
    <div
      class="input-box"
    >
      <input
        required="required"
        type="text"
      />
       
      <label>
        Selling Commission
      </label>
    </div>
     
    <div
      class="input-box margin"
    >
      <div
        class="selector"
      >
        <div
          class="label"
        >
          Margin
        </div>
         
        <div
          class="switch"
        >
          <input
            class="switch-input"
            type="checkbox"
          />
           
          <label
            class="switch-label"
          />
        </div>
      </div>
    </div>
     
    <div
      class="show-margin"
    >
      <div
        class="input-box"
      >
        <input
          required="required"
          type="text"
        />
         
        <label>
          Interest Rate (%)
        </label>
      </div>
       
      <div
        class="input-box"
      >
        <input
          required="required"
          type="text"
        />
         
        <label>
          Initial Deposit (%)
        </label>
      </div>
       
      <div
        class="input-box"
      >
        <input
          required="required"
          type="text"
        />
         
        <label>
          Duration (Day)
        </label>
      </div>
       
      <div
        class="input-box result"
      >
        <span
          class="tag"
        >
          Cash Required:
        </span>
         
        <span>
          $0.00
        </span>
      </div>
    </div>
     
    <div
      class="divider"
    />
     
    <div
      class="input-box result"
    >
      <span
        class="tag"
      >
        Cost:
      </span>
       
      <span>
        $0.00
      </span>
    </div>
     
    <div
      class="input-box result"
    >
      <span
        class="tag"
      >
        • Fee & Commission:
      </span>
       
      <span>
        $0.00
      </span>
    </div>
     
    <div
      class="input-box result"
      style="display: none;"
    >
      <span
        class="tag"
      >
        • Margin Interest:
      </span>
       
      <span>
        $0.00
      </span>
    </div>
     
    <div
      class="input-box result"
    >
      <span
        class="tag"
      >
        Revenue:
      </span>
       
      <span>
        $0.00
      </span>
    </div>
     
    <div
      class="input-box result"
    >
      <span
        class="tag"
      >
        • Fee & Commission:
      </span>
       
      <span>
        $0.00
      </span>
    </div>
     
    <div
      class="input-box result"
    >
      <span
        class="tag"
      >
        Profit:
      </span>
       
      <span>
        $0.00
      </span>
    </div>
     
    <div
      class="input-box result"
    >
      <span
        class="tag"
      >
        Percentage:
      </span>
       
      <span>
        0.00%
      </span>
    </div>
  </div>
   
  <div
    class="footer"
  >
    <div
      class="disclaimer"
    >
      
    Disclaimer: The calculation is for reference only.
  
    </div>
  </div>
</div>
`;
